package.json

```diff
@@ -74,11 +74,11 @@
 		"sentences-per-line": "^0.2.1",
 		"should-semantic-release": "^0.0.4",
 		"tsup": "^6.5.0",
-		"typescript": "4.9.5",
+		"typescript": "5.0.0-beta",
 		"vitest": "^0.28.0"
 	},
 	"peerDependencies": {
-		"typescript": ">=4"
+		"typescript": ">=5 || 5.0.0-beta"
```

Comment:
In theory, I love this - easier imports, lots of removed functions, nice clean minimum TypeScript version.

In practice, https://github.com/microsoft/TypeScript/issues/51362 notes that TS v5 comes out in a month, on March 14th. So folks will be in a hard position till then: either they stay on `tsutils`, stay on an older version of `ts-api-utils`, or use a beta version of TypeScript. What do you think about waiting till then @RebeccaStevens? 

cc @bradzacher from https://github.com/typescript-eslint/typescript-eslint/issues/6380 & https://github.com/typescript-eslint/typescript-eslint/pull/6459, would the typescript@5 requirement be inconvenient to you?

Reply:
To me, waiting a month before making a 1.0 release is no big deal - it gives us plenty of time to get this library right. I guess it really just depends on when typescript-eslint want to release v6.

Reply:
Making it the minimum would prevent us from using it really because we support older TS versions.

Reply:
We don't actually rely on any v5 things so I guess we don't need to set the minimum version to v5.
We can just design the utility functions around v5 and then add any v4 one needed by typescript-eslint or other dependents. Or if we re-export/redefine type guards already defined in TypeScript (#51), then we can easily support older version of TypeScript.

Reply:
For context as of our upcoming v6 release (which is when we'll switch to this package) we'll be bumping the minimum TS version to 4.2 - which is the same minimum that DefinitelyTyped supports.

So we could do the same thing here and keep everything perfectly balanced, as all things should be.

Reply:
I like keeping the minimum TS version synced to DefinitelyTyped and typescript-eslint, if these can all be the same.

Reply:
In order to maintain safety in the codebase we use `patch-package` to edit the TS types to help flag when we use things that aren't in older versions. I think it's a great strategy we could use here if needs be.